Create specs for stores pages (#263)
PrestaShop/prestashop-specs: content/1.7/back-office/shop-parameters/contact/stores/add-edit stores.md

---

> title: Add / Edit Stores

Reviewer: ```suggestion
title: Add / Edit Store
```

weight: 1
---

# Add / Edit Store

While adding or editing a store, the following fields/toggles/drop downs/parameters are displayed:

1) Name

> This input field allows to fill the name of the store. It can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down. When hovered on "name", shows the following suggestion: "Store name (e.g. City Center Mall Store). Allowed characters: letters, spaces and %s"

Reviewer: ```suggestion
This input field allows to fill the name of the store. It can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down. When the label is hovered, a tooltip is displayed: "Store name (e.g. City Center Mall Store). Allowed characters: letters, spaces and %s"
```

2) Address

This input field allows to fill the address of the store. This field is mandatory and can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.
If the save button is pressed and if this field is empty, the following error message is displayed: "The field address1 is required at least in _default language_"

3) Address (2)

This input field allows to fill the second address line of the store. This field can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.

4) Zip/Postal code

> This input field allows to fill the zip/postal code of the store. The field is not marked as mandatory, but if the field is empty, and trying to save, shows the message: "A Zip/Postal code is required.". Also the postcode can't be too long, if it's too long, and trying to save shows the message: "The postcode field is too long (12 chars max).", also if the format isn't correct, when saving shows error: "The Zip/Postal code is invalid. It must be entered as follows: 00000"

Reviewer: ```suggestion
This input field allows to fill the zip/postal code of the store. The field is not marked as mandatory, but if the field is empty when trying to save, the following error message is displayed: "A Zip/Postal code is required.". If the postal code exceeds 12 characters when trying to save, the following error message is displayed: "The postcode field is too long (12 chars max).". If the format isn't correct when saving, the following error message is displayed: "The Zip/Postal code is invalid. It must be entered as follows: 00000". The format of the postal code is defined for each country in International > Locations > Countries > Add/edit country.
```

Reviewer: I added missing information here about the format of the postal code: it's defined for each country in International > Locations > Countries > Add/edit country.

5) City

This input field allows to fill the city of the store and is mandatory.
If the save button is pressed and if this field is empty, the following error message is displayed: "The city field is required."

6) Country

This drop-down allows to select the country of the store and is mandatory. While creating a new store, the default country (selected in International > Localization > Configuration) is selected by default.

> 7) State (only for the countries that have states, as defined in International > Locations > States.)

Reviewer: ```suggestion
7) State (displayed only for the countries that have states, as defined in International > Locations > States.)
```

> This drop-down allows to select the state of the store and is mandatory. If no state is selected, when trying to save, shows: "An address located in a country containing states must have a state selected."

Reviewer: ```suggestion
This drop-down allows to select the state of the store and is mandatory. If no state is selected when trying to save, the following error message is displayed: "An address located in a country containing states must have a state selected."
```

8) Latitude/Longtitude

There are 2 separate input fields for the latitude and longitude of the store, they are both mandatory.

> Only numbers are allowed for these fields. When trying to input characters that are not some numbers, an error message is displayed: "Latitude and longitude are required." The numbers in the fields are limited to 13, if there are more numbers than allowed, an error message is displayed: "The latitude field is too long (13 chars max)." and/or "The longitude field is too long (13 chars max)."

Reviewer: ```suggestion
Only numbers are allowed for these fields. When trying to input a character that is not a number, an error message is displayed: "Latitude and longitude are required." The numbers in the fields are limited to 13, if there are more numbers than allowed, an error message is displayed: "The latitude field is too long (13 chars max)." and/or "The longitude field is too long (13 chars max)."
```

When hovered on the writing - Latitude / Longitude, shows the following: "Store coordinates (e.g. 45.265469/-47.226478).

9) Phone

This input field allows to fill the phone number of the store.

10) Fax

This input field allows to fill the fax number of the store.

11) Email address

This input field allows to fill the email address of the store.

12) Note

> This input field allows to fill a note for the store. This field can be translated, a drop-down list is shown, translates the field to a selected language which are added in the shop.

Reviewer: ```suggestion
This input field allows to fill a note for the store. It can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.
```

13) Active

> This toggle can be on "yes" or "no" and allows to choose if the store is active or not. Default state of this button when created is "yes"

Reviewer: ```suggestion
This toggle can be on "yes" or "no" and allows to choose if the store is active or not. By default, while creating a new store, this toggle is on "yes". If the toggle is set to "no", the "Stores" page of the front office will not be displayed or accessible.
```

> When hovered on "Active", shows the following: "Whether or not to display this store". If the button is set to "no", in the front office, Stores section the shop will not be displayed.

Reviewer: ```suggestion
When the label is hovered, a tooltip is displayed: "Whether or not to display this store". If the toggle is set to "no", the "Stores" page of the front office will not be displayed or accessible.
```

14) Picture

> Picture is shown that will be used for the store.

Reviewer: ```suggestion
This field allows to upload a picture for the store.
```

While editing a store with an existing picture, the size of the picture is displayed under the field. When hovered on Picture, shows the following: "Storefront picture."

> When trying to upload an incorrect format file, shows the message: "Image format not recognized, allowed formats are: .gif, .jpg, .png"

Reviewer: ```suggestion
When trying to upload an incorrect format file, the following error message is displayed: "Image format not recognized, allowed formats are: .gif, .jpg, .png"
```

---
title: Add / Edit Stores
weight: 1
---

# Add / Edit Store

When entered the add or editing current store the following fields are displayed:

1) Name

> Input field for the name of the store. Can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.

Reviewer: When the label is hovered, a tooltip is displayed. I let you specify this.

> 

Reviewer: When the label is hovered, a tooltip is displayed. I let you specify this.

3) Address

Input field for the address of the shop. This field is mandatory and can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.
If there is no address input, when trying to save shows the message: "The field address1 is required at least in English (English)."

5) Address (2)

Second input field for the address of the shop. This field can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the field is displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.

7) Zip/Postal code

Input field for the zip/postal code of the shop.

> 

Reviewer: An error message is displayed if the postal code is empty or not in the right format, I let you specify this.

9) City

Input field for the city of the shop, this field is mandatory.
When trying to save without a city entered, shows the message: "The city field is required."

11) Country

Drop-down menu for the country of the shop, this is a mandatory selection field.

13) State (is shown only when the country is selected as "United states")

Drop-down menu for the state of the shop, this is a mandatory selection field.

> 

Reviewer: An error message is displayed if no state is selected, I let you specify this.

15) Latitude/Longtitude

These are 2 separate input fields for the lattitude and longtitude of the shop, this field is mandatory.
Only numbers are allowed for these fields. When trying to input anything but not numbers, a message is shown: "Latitude and longitude are required." Also, the numbers in the fields are limited, if there are more numbers than allowed, messaged are shown: 
"There are 2 errors.
The latitude field is too long (13 chars max).
The longitude field is too long (13 chars max)."

17) Phone

Input field for the phone of the shop.

19) Fax

Input field for the fax of the shop.

21) Email address

Input field for the email of the shop.

23) Note

Input field for the notes of the shop.

> 

Reviewer: This field can be translated

25) Active

Toggle switch for "yes" or "no"

> 

Reviewer: What is the default state while adding a store ? Is it on Yes or No ?
What happens in front-office if it's not active ?

When the label is hovered, a tooltip is displayed. I let you specify this.

27) Picture

Picture is shown that will be used for the store.
Below the size of the picture is shown.
Below that there is a field for picture name input, and a button "add file" for adding a picture.

29) Shop association

The shop association block is displayed when the multistore is enabled and if several shops are configured.

:point_right: See the [specs of the shop association block](../../../multistoregeneralspecs.md#shop-association-block)

31) Hours

Firstly an example is shown: "e.g. 10:00AM - 9:30PM"

> All the fields below have separate inputs for working hours. By default all of them are fields have already inputs "09:00AM - 07:00PM" that can be changed. These fields can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the fields are displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.

Reviewer: ```suggestion
For each day of the week, there is an input field allowing to fill the working hours. These fields can be translated: When several languages are installed, a drop-down next to the field is displayed. It allows choosing in which language the fields are displayed and customizing it according to the language. Disabled languages are displayed in the drop-down.
```
